pennylane/transforms/core/transform_dispatcher.py

```diff
@@ -425,17 +425,58 @@ def default_qnode_transform(self, qnode, targs, tkwargs):
         return qnode
 
 
-class TransformContainer:  # pylint: disable=too-many-instance-attributes
-    """Class to store a quantum transform with its ``args``, ``kwargs`` and classical co-transforms.  Use
-    :func:`~.pennylane.transform`.
+class BoundTransform:  # pylint: disable=too-many-instance-attributes
+    """A transform with bound inputs.
 
-    .. warning::
+    Args:
+        transform: Any transform.
+        args (Sequence[Any]): The positional arguments to use with the transform.
+        kwargs (Dict | None): The keyword arguments for use with the transform.
 
-        This class is developer-facing and should not be used directly. Instead, use
-        :func:`qml.transform <pennylane.transform>` if you would like to make a custom
-        transform.
+    Keyword Args:
+        use_argnum (bool): An advanced option used in conjunction with calculating
+            classical cotransforms of jax workflows.
 
     .. seealso:: :func:`~.pennylane.transform`
+
+    >>> bound_t = BoundTransform(qml.transforms.merge_rotations, (), {"atol": 1e-4})
+    >>> bound_t
+    <merge_rotations((), {'atol': 0.0001})>
+
+    These objects can now directly applied to anything individual transforms can apply to:
+
+    .. code-block:: python
+
+        @bound_t
+        @qml.qnode(qml.device('null.qubit'))
+        def c(x):
+            qml.RX(x, 0)
+            qml.RX(-x + 1e-6, 0)
+            qml.RY(x, 1)
+            qml.RY(-x + 1e-2, 1)
+            return qml.probs(wires=(0,1))
+
+    If we draw this circuit, we can see that the ``merge_rotations`` transforms was applied with a
+    tolerance of ``1e-4``.  The ``RX`` gates sufficiently close to zero disappear, while the ``RY`` gates
+    that are further from zero remain.
+
+    >>> print(qml.draw(c)(1.0))
+    0: ───────────┤ ╭Probs
+    1: ──RY(0.01)─┤ ╰Probs
+
+    Repeated versions of the bound transform can be created with multiplication:
+
+    >>> bound_t * 3
+    TransformProgram(merge_rotations, merge_rotations, merge_rotations)
+
+    And it can be used in conjunction with both individual transforms, bound transforms, and
+    transform programs.
+
+    >>> bound_t + qml.transforms.cancel_inverses
+    TransformProgram(merge_rotations, cancel_inverses)
+    >>> bound_t + qml.transforms.cancel_inverses + bound_t
+    TransformProgram(merge_rotations, cancel_inverses, merge_rotations)
```

Comment:
```suggestion
    CompilePipeline(merge_rotations, merge_rotations, merge_rotations)

    And it can be used in conjunction with both individual transforms, bound transforms, and
    transform programs.

    >>> bound_t + qml.transforms.cancel_inverses
    CompilePipeline(merge_rotations, cancel_inverses)
    >>> bound_t + qml.transforms.cancel_inverses + bound_t
    CompilePipeline(merge_rotations, cancel_inverses, merge_rotations)
```